### ISSUE TITLE
New version of Quarto not respecting render order

### ISSUE THREAD

**Issue by maxdrohde** [quarto-dev/quarto-cli#10490]
### Bug description

I have been using Quarto to render a series of `.qmd` files in a certain order, where each Quarto file produces an output that is then used by the next Quarto file. For example:

```
project:
  title: "VIOLET Analysis"
  output-dir: _output
  render:
    - 1-clean_data.qmd
    - 2-data_summary.qmd
    - 3-create_eda_figures.qmd
    - 4-model_fitting.qmd
    - 5-model_analysis.qmd
    - 6-derived_quantities.qmd
    - 7-sops.qmd
```

In the past, the files always ran in the order that I specified in the YAML list. However, when I ran my code again with `quarto render` from the command line using the latest version of Quarto (1.5.56), file 2 ran before file 1. When I downgraded to Quarto 1.3.450, the scripts ran in order again.

### Steps to reproduce

I had trouble creating a minimal example, but you can see the difference using this Quarto project in my GitHub repository:

<https://github.com/maxdrohde/bayesian_ordinal_transition/tree/main/violet>

Running `quarto render` with Quarto 1.5.56 runs file 2 before file 1, but using Quarto 1.3.450, the files run in order.

### Expected behavior

Files should run in the order specified by `render:` in the YAML header.

### Actual behavior

Files run out of order.

### Your environment

MacOS: 14.5 (23F79)

### Quarto 1.3.450

```
❯ quarto check

[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.1: OK
      Dart Sass version 1.55.0: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 1.3.450
      Path: /Applications/quarto/bin

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.11.3
      Path: /Library/Frameworks/Python.framework/Versions/3.11/bin/python3
      Jupyter: 5.3.0
      Kernels: ir, julia-1.6, julia-1.8, julia-1.9, juliapro_v1.5.0-1-1.5, julia-1.5, julia-12-threads-1.8, python3

(|) Checking Jupyter engine render....0.00s - Debugger warning: It seems that frozen modules are being used, which may
0.00s - make the debugger miss breakpoints. Please pass -Xfrozen_modules=off
0.00s - to python to disable frozen modules.
0.00s - Note: Debugging will proceed. Set PYDEVD_DISABLE_FILE_VALIDATION=1 to disable this validation.
0.00s - Debugger warning: It seems that frozen modules are being used, which may
0.00s - make the debugger miss breakpoints. Please pass -Xfrozen_modules=off
0.00s - to python to disable frozen modules.
0.00s - Note: Debugging will proceed. Set PYDEVD_DISABLE_FILE_VALIDATION=1 to disable this validation.
[✓] Checking Jupyter engine render....OK

[✓] Checking R installation...........OK
      Version: 4.4.0
      Path: /Library/Frameworks/R.framework/Resources
      LibPaths:
        - /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/library
      knitr: 1.48
      rmarkdown: 2.27

[✓] Checking Knitr engine render......OK
```

### Quarto 1.5.56

```
❯ quarto check
Quarto 1.5.56
[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.2.0: OK
      Dart Sass version 1.70.0: OK
      Deno version 1.41.0: OK
      Typst version 0.11.0: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 1.5.56
      Path: /Applications/quarto/bin

[✓] Checking tools....................OK
      TinyTeX: (not installed)
      Chromium: (not installed)

[✓] Checking LaTeX....................OK
      Using: Installation From Path
      Path: /Library/TeX/texbin
      Version: 2023

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.11.3
      Path: /Library/Frameworks/Python.framework/Versions/3.11/bin/python3
      Jupyter: 5.3.0
      Kernels: ir, julia-1.6, julia-1.8, julia-1.9, juliapro_v1.5.0-1-1.5, julia-1.5, julia-12-threads-1.8, python3

[✓] Checking Jupyter engine render....OK

[✓] Checking R installation...........OK
      Version: 4.4.0
      Path: /Library/Frameworks/R.framework/Resources
      LibPaths:
        - /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/library
      knitr: 1.48
      rmarkdown: 2.27

[✓] Checking Knitr engine render......OK
```

**Comment by allefeld** [comment]
I have the same issue with 1.5.56. The order under `project / render` is no longer respected.

**Comment by maxdrohde** [comment]
Fixed with #11062
